apollo-router/src/warp_http_server_factory.rs

```diff
@@ -676,4 +692,36 @@ mod tests {
             .iter()
             .for_each(|accepts| assert!(!prefers_html(accepts.to_string())));
     }
+
+    #[test(tokio::test)]
+    async fn it_provides_health_status_body() -> Result<(), FederatedServerError> {
+        let filter = get_health_request();
+
+        let res = warp::test::request()
+            .path("/.well-known/apollo/server-health")
+            .reply(&filter)
+            .await;
+
+        assert_eq!(res.status(), 200);
+        assert_json_snapshot!(String::from_utf8_lossy(res.body()));
+
+        Ok(())
+    }
+
+    #[test(tokio::test)]
+    async fn it_provides_health_status_header() -> Result<(), FederatedServerError> {
+        let filter = get_health_request();
+
+        let res = warp::test::request()
+            .path("/.well-known/apollo/server-health")
+            .reply(&filter)
+            .await;
+
+        let hdrs = res.headers();
+
+        assert_eq!(res.status(), 200);
+        assert_snapshot!(hdrs["content-type"].to_str().unwrap());
```

Comment:
[nit] Try using assert_debug_snapshot on the entire `res` object. Much easier!!

Reply:
I only want to verify the content-type header is in place. If there is a different (new) header in the response, I don't want my test to start to fail.

Reply:
It's using `warp::test::request()` so I suspect it won't have too many changing headers. I have not tested though.

Even if something change in the future that make the response look different, the test will fail and display the difference and someone will just valid the new snapshot. No biggy.